simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -36,96 +37,119 @@
-        switch (simulatorMode) {
-            case PUBLISHER -> simulatorModeHandler = new PublisherModeHandler(
+        this.simulatorModeHandler = switch (simulatorMode) {
+            case PUBLISHER -> new PublisherModeHandler(
                     blockStreamConfig, publishStreamGrpcClient, blockStreamManager, metricsService);
-            case CONSUMER -> simulatorModeHandler = new ConsumerModeHandler(blockStreamConfig);
-            case BOTH -> simulatorModeHandler = new CombinedModeHandler(blockStreamConfig);
+            case CONSUMER -> new ConsumerModeHandler(blockStreamConfig, consumerStreamGrpcClient);
+            case BOTH -> new CombinedModeHandler(blockStreamConfig);
             default -> throw new IllegalArgumentException("Unknown SimulatorMode: " + simulatorMode);
-        }
+        };
     }
 
     /**
-     * Starts the block stream simulator.
+     * Initializes and starts the block stream simulator in the configured mode.
+     * This method initializes all components and begins the streaming process.
      *
-     * @throws InterruptedException if the thread is interrupted
-     * @throws BlockSimulatorParsingException if a parse error occurs
-     * @throws IOException if an I/O error occurs
+     * @throws InterruptedException if the streaming process is interrupted
+     * @throws BlockSimulatorParsingException if a block parsing error occurs
+     * @throws IOException if an I/O error occurs during streaming
      */
     public void start() throws InterruptedException, BlockSimulatorParsingException, IOException {
-        LOGGER.log(System.Logger.Level.INFO, "Block Stream Simulator started initializing components...");
-        publishStreamGrpcClient.init();
+        LOGGER.log(INFO, "Block Stream Simulator started initializing components...");
+        simulatorModeHandler.init();
+
         isRunning.set(true);
 
         simulatorModeHandler.start();
     }
 
     /**
-     * Returns whether the block stream simulator is running.
+     * Checks if the simulator is currently running.
      *
-     * @return true if the block stream simulator is running, false otherwise
+     * @return true if the simulator is running, false otherwise
      */
     public boolean isRunning() {
         return isRunning.get();
     }
 
     /**
-     * Stops the Block Stream Simulator and closes off all grpc channels.
+     * Gracefully stops the simulator and closes all gRPC channels.
+     * This method ensures proper cleanup of resources and termination of streaming operations.
      *
-     * @throws InterruptedException if the thread is interrupted
+     * @throws InterruptedException if the shutdown process is interrupted
      */
     public void stop() throws InterruptedException {
         simulatorModeHandler.stop();
-        publishStreamGrpcClient.completeStreaming();
-
-        publishStreamGrpcClient.shutdown();
         isRunning.set(false);
```

Comment:
I believe we have made in the past an issue to add different statuses for the simulator in order to have more granularity in determining if the app is shutting down or is shut down or is initializing or running or paused etc. etc.
If that is so maybe we could tag this `stop` method with the `@todo(issueNumber)` convention?

Reply:
Added this to this method, no need to spam all the places, issue is in the backlog and is visible https://github.com/hashgraph/hedera-block-node/issues/322